astropy/units/equivalencies.py

```diff
@@ -58,12 +70,17 @@ class Equivalency(list):
         Any positional or keyword arguments used to make the equivalency.
     """
 
-    def __init__(self, equiv_list, name="", kwargs=None):
+    def __init__(
+        self,
+        equiv_list: list[EquivalencyPair],
+        name: str = "",
+        kwargs: Mapping[str, object] | None = None,
+    ) -> None:
         super().__init__(equiv_list)
         self.name = [name]
         self.kwargs = [kwargs] if kwargs is not None else [{}]
 
-    def __add__(self, other):
+    def __add__(self, other: list[Any]) -> list[Any]:
```

Comment:
This might not be the best annotation, but I couldn't come up with overloads that agree with `list.__add__()`, so I decided this is good enough.